open_flamingo/eval/evaluate.py

```diff
@@ -296,94 +344,98 @@ def get_random_indices(num_samples, query_set_size, full_dataset, seed):
-    get_prompt: Callable[[dict], str],
-    in_context_samples,
-    effective_num_shots,
-    num_shots,
-) -> str:
-    context_text = (
-        "".join([get_prompt(s) for s in in_context_samples])
-        if effective_num_shots > 0
-        else ""
-    )
+def get_query_set(train_dataset, query_set_size, seed):
+    np.random.seed(seed)
+    query_set = np.random.choice(len(train_dataset), query_set_size, replace=False)
+    return [train_dataset[i] for i in query_set]
 
-    if num_shots == 0:
-        context_text = context_text.replace("<image>", "")
-    return context_text
+
+def prepare_eval_samples(test_dataset, num_samples, seed):
+    np.random.seed(seed)
+    random_indices = np.random.choice(len(test_dataset), num_samples, replace=False)
+    return torch.utils.data.Subset(test_dataset, random_indices)
 
 
 def sample_batch_demos_from_query_set(query_set, num_samples, batch_size):
     return [random.sample(query_set, num_samples) for _ in range(batch_size)]
 
 
-def evaluate_coco_flickr(
+def evaluate_captioning(
+    args,
     eval_model,
-    batch_size,
-    image_dir_path,
-    annotations_json_path,
     seed=42,
     max_generation_length=20,
     num_beams=3,
     length_penalty=-2.0,
-    num_samples=5000,
     query_set_size=2048,
     num_shots=8,
-    is_flickr=False,
+    dataset_name="coco",
 ):
     """Evaluate a model on COCO dataset.
 
     Args:
+        args (argparse.Namespace): arguments
         eval_model (eval_model.EvalModel): model to evaluate
-        batch_size (int): batch size
-        image_dir_path (str, optional): path to the directory containing the images.
-        annotations_json_path (str, optional): path to the json file containing the annotations.
         seed (int, optional): seed for random number generator. Defaults to 42.
         max_generation_length (int, optional): maximum length of the generated caption. Defaults to 10.
         num_beams (int, optional): number of beams to use for beam search. Defaults to 3.
         length_penalty (float, optional): length penalty for beam search. Defaults to -2.0.
-        num_samples (int, optional): number of samples to evaluate on. Defaults to 5000.
         query_set_size (int, optional): number of samples to use for query set. Defaults to 2048.
         num_shots (int, optional): number of in-context samples to use. Defaults to 8.
         num_workers (int, optional): number of workers to use for dataloader. Defaults to 4.
-        is_flickr (bool): defines if that data is COCO or Flickr. Defaults to False (COCO).
-
+        dataset_name (str, optional): dataset to evaluate on. Defaults to "coco". Can be "coco" or "flickr".
     Returns:
         float: CIDEr score
 
     """
 
-    full_dataset = COCOFlickrDataset(
-        image_dir_path=image_dir_path,
-        annotations_path=annotations_json_path,
-        is_flickr=is_flickr,
+    train_dataset = CaptionDataset(
+        image_train_dir_path=args.coco_train_image_dir_path
```

Comment:
is there a way to clean this up without 3 if/else blocks that depend on the same variable?